crates/rune/src/runtime/bytes.rs

```diff
@@ -6,13 +6,15 @@ use crate::compile::{InstallWith, Named};
 use crate::runtime::{
     FromValue, Mut, RawMut, RawRef, RawStr, Ref, UnsafeFromValue, Value, VmError,
 };
+use serde::{Deserialize, Serialize};
 use std::cmp;
 use std::fmt;
 use std::ops;
 
 /// A vector of bytes.
-#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Deserialize, Serialize)]
```

Comment:
Since `Bytes` is purely a container of bytes it's more appropriate that it gets serialized transparently as an array of bytes, rather than deriving `Serialize` on the struct which has a field `inner` that is serialized as bytes.

A way to achieve this would be to implement `Serialize` / `Deserialize` directly and call the appropriate serialization / deserialization functions. Alternatively I think you can add `#[serde(transparent)]` to this struct.